### PR TITLE
bug: fix CMC routes

### PR DESCRIPTION
**Pull request by Jaguar0625**
- total: all minted funds (initial + inflation)

- circulating: total - burned (returned to nemesis)

- max: maximum minted funds

### CODE REVIEW

**Comment by 0x6861746366574** [comment]
Given the sinkAddresses are currently considered reserved, can we remove `NCVORTEX4XD5IQASZQEHDWUXT33XBOTBMKFDCLI` and `NAVORTEX3IPBAUWQBBI3I3BDIOS4AVHPZLCFC7Y` from circulatingSupply, please @Jaguar0625?

File: rest/test/plugins/cmc/cmcRoutes_spec.js [hunk 1]
```diff
@@ -27,18 +28,22 @@ const { expect } = require('chai');
 const sinon = require('sinon');
 const fs = require('fs');
 
-const { uint64 } = catapult.utils;
-
 describe('cmc routes', () => {
 	describe('network currency supply', () => {
 		const maxSupply = 9000000000000000;
-		const XYMSupply = 8998999998000000;
+		const xymSupply = 8998999998000000;
```
Comment (0x6861746366574) [inline rest/test/plugins/cmc/cmcRoutes_spec.js:34]: Even though this is a test, should this not be `8'999'999'999'000'000`?

**Review by 0x6861746366574 — CHANGES_REQUESTED** [review]
Should we change `cmc` to `verifySupply` or `supplyVerification` or `xymSupplyRoutes`? Our supply endpoints shouldn't be attached to a third-party service.

File: rest/resources/rest.json [hunk 2]
```diff
@@ -12,6 +12,10 @@
     "allowedHosts": ["*"],
     "allowedMethods": ["GET", "POST", "PUT", "OPTIONS"]
   },
+  "uncirculatingAccountPublicKeys": [
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply rest/resources/rest.json:15]: i would not - this is more general, so we don't need to modify code when/if we change wallet structure

File: rest/src/plugins/cmc/cmcRoutes.js [hunk 3]
```diff
@@ -42,42 +42,42 @@ module.exports = {
...
 			}));
 
 		server.get('/network/currency/supply/total', (req, res, next) => readAndParseNetworkPropertiesFile()
 			.then(propertiesObject => {
-				const currencyId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
-				return db.mosaicsByIds([currencyId]).then(response => {
-					const supply = response[0].mosaic.supply.toString();
+				const currencyMosaicId = propertyValueToMosaicId(propertiesObject.chain.currencyMosaicId);
+				return db.mosaicsByIds([currencyMosaicId]).then(response => {
+					const supply = parseInt(response[0].mosaic.supply.toString(), 10);
```
Comment (gimre-xymcity) [inline rest/src/plugins/cmc/cmcRoutes.js]: why is there .toString and parseInt needed here? o0
`mosaic.supply` here is hex string or something else?

Comment (Jaguar0625) [reply rest/src/plugins/cmc/cmcRoutes.js]: it is MongoDb.Long

Comment (Jaguar0625) [reply rest/src/plugins/cmc/cmcRoutes.js]: although that's from an old revision, and that should be removed

File: rest/resources/rest.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (0x6861746366574) [reply rest/resources/rest.json:15]: Mmm. Alright.

**Review by 0x6861746366574 — APPROVED** [review]

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**